### PR TITLE
Reorganizing examples

### LINKED ISSUE #563: Reorganize examples
Currently, we are dropping new examples into the `examples` directory. Since the directory is getting larger, it is harder to orient there. I received the first feedback already that people had different expectations when looking at the examples. Also, the users may want to look for specific kinds of examples. I am proposing the following structure:

 - `language` for simple examples that demonstrate some basic language features. Now we have plenty of small files right in `./examples`, which is not good.
 - `distributed` for distributed algorithms containing:
   - `LamportMutex`
   - `ClockSync`
   - `Paxos`
   - `ReadersWriters`
   - `ewd840`
   - `tendermint`
 - `cosmos` for cosmos modules, dApps, and smart contracts containing:
   - `ics23`
   - `bank` (currently, in a separate branch)
 - `solidity` for Solidity examples. Although it's not our primary goal to support Solidity, there is a large set of well-scoped Solidity contracts, which could work well as introductory examples. Also, Evmos is using Solidity, so it's related to Cosmos. Currently, we have:
   - `ERC20`
 - `puzzles` for small examples that are good for learning combinatorial behavior of model checkers, though they may be misleading, as combinatorial problems are often harder than e.g. smart contracts. Currently, we have:
   - `tictactoe`
 - `sequential` for sequential algorithms. Again, they are not our primary goal but they may help the users to connect better with the solutions they already know.

@bugarela, @shonfeder, @thpani, @Kukovec, ideas?

### PR DESCRIPTION
**Pull request by konnov**
Closes #563. As discussed there, reorganizing examples by categories, so it is easier to find the examples certain audiences may be interested in.

- [x] Entries added to the respective `CHANGELOG.md` for any new functionality

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -13,6 +13,10 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
  - Add an example that specifies the Solidity Coin contract (#576)
  - A work-in-progress example on Solidity's Simple Auction (#573)
 
+### Changed
+
+ - `examples` are grouped by categories (#582)
+
```
Comment (shonfeder) [inline CHANGELOG.md:19]: I don't think we should add changelog entries for doc organization changes. It's not part of the software or api afaiu.

Comment (shonfeder) [inline CHANGELOG.md:19]:
```suggestion
```

**Review by shonfeder — APPROVED** [review]

**Comment by shonfeder** [comment]
What about the readme thomas suggested?

**Comment by konnov** [comment]
> What about the readme thomas suggested?

Good point. I will open a follow-up issue, so this one can be merged right away.

**Pull request merged**